library/src/main/java/com/danikula/videocache/HttpUrlSource.java

```diff
@@ -91,6 +95,10 @@ public int read(byte[] buffer) throws ProxyCacheException {
             throw new InterruptedProxyCacheException("Reading source " + url + " is interrupted", e);
         } catch (IOException e) {
             throw new ProxyCacheException("Error reading data from " + url, e);
+        } catch (NullPointerException e) {
+            throw new ProxyCacheException("Error reading data with NullPointerException from " + url, e);
+        } catch (Exception e) {
```

Comment:
I dan't want to catch any Exception including all Runtime exceptions. It can hide real problems. Please remove this catch block.